pmd-cataloger/src/main/java/sfdc/sfdx/scanner/messaging/SfdxMessager.java

```diff
@@ -36,64 +23,43 @@ public static SfdxMessager getInstance() {
     return INSTANCE;
   }
 
-  public void uxInfo(boolean verbose, EventKey key, String... args) {
-    uxAddMessage(
-      verbose,
-      MessageType.INFO,
-      key,
-      args);
-  }
-
-  public void uxWarn(boolean verbose, EventKey key, String... args) {
-    uxAddMessage(
-      verbose,
-      MessageType.WARNING,
-      key,
-      args);
-  }
-
-  public void uxError(SfdxScannerException exception) {
-    uxAddMessage(
-      false,
-      MessageType.ERROR,
-      MessageHandler.UX,
+  /**
+   * Add exception to pass onto Typescript layer.
+   * Will be treated as an Error based on the properties set
+   * in EventKey. Please make sure that EventKey is correct and is
+   * in sync with messages/EventKeyTemplates.json
+   * @param exception to send to Typescript layer
+   */
+  public void addMessage(SfdxScannerException exception) {
+    final EventKey eventKey = exception.getEventKey();
+    addMessage(
       exception.getFullStacktrace(),
-      exception.getEventKey(),
+      eventKey,
       exception.getArgs());
   }
 
-  private void uxAddMessage(
-    boolean verbose,
-    MessageType messageType,
-    EventKey key,
-    String... args) {
-    uxAddMessage(
-      verbose,
-      messageType,
-      MessageHandler.UX,
-      "",
-      key,
-      args);
-  }
-
-  private void uxAddMessage(
-    boolean verbose,
-    MessageType messageType,
-    MessageHandler messageHandler,
-    String log,
-    EventKey eventKey,
-    String... args) {
+  /**
+   * Add message to pass onto Typescript layer.
+   * Make sure EventKey is updated with messages/EventKeyTemplates.json
+   * and has correct properties in the enum.
+   * @param log Information for internal use. Will be logged but not displayed to user
+   * @param eventKey EventKey to display to user
+   * @param args String args passed to the EventKey to make the displayed message meaningful
+   */
+  public void addMessage(String log, EventKey eventKey, String... args) {
```

Comment:
Why do we need `log` to be a separate string from what we display to the user? Can't we just log the same thing we display? If we feel the need to make them separate, I'd rename the `log` parameter to something like `internalLog`, to make it extra clear that it is completely separate from what gets displayed to the user.

Reply:
`log` is to keep additional free style information that can help with debugging, but are not essential to display to users. The name `internalLog` sounds even better. Let me change it.